typescript/api/controllers/BrandingAppController.ts

```diff
@@ -0,0 +1,119 @@
+ */
+import type { Request, Response } from 'sails';
+import { Controllers as controllers } from '@researchdatabox/redbox-core-types';
+
+declare const sails: any;
+declare const BrandingService: any;
+declare const BrandingLogoService: any;
+declare const BrandingConfig: any;
+declare const BrandingConfigHistory: any;
+
+function mapError(e: Error): { status: number; body: any } {
+  const msg = e.message || '';
+  if (msg === 'unauthorized') return { status: 403, body: { error: 'forbidden' } };
+  if (msg.startsWith('Invalid variable key')) return { status: 400, body: { error: 'invalid-variable', detail: msg } };
+  if (msg.startsWith('contrast-violation')) return { status: 400, body: { error: 'contrast', detail: msg } };
+  if (msg === 'branding-not-found') return { status: 404, body: { error: 'branding-not-found' } };
+  if (msg === 'history-not-found') return { status: 404, body: { error: 'history-not-found' } };
+  if (msg === 'publish-conflict') return { status: 409, body: { error: 'publish-conflict' } };
+  if (msg.startsWith('logo-invalid')) return { status: 400, body: { error: 'logo-invalid', detail: msg } };
+  return { status: 500, body: { error: 'server-error', detail: msg } };
+}
+
+// Authorization for these endpoints is now handled via the configured policy rules
+// in config/auth.js (pattern: /:branding/:portal/app/branding/*). We only map errors here.
+
+export module Controllers {
+  export class BrandingApp extends controllers.Core.Controller {
+    protected _exportedMethods: any = ['config','draft','preview','publish','logo'];
+
+    /** 9.1 Return current draft/active branding config + logo meta */
+    async config(req: Request, res: Response) {
+      try {
+        const branding = req.params['branding'];
+        const brand = await BrandingConfig.findOne({ name: branding });
+        if (!brand) return res.status(404).json({ error: 'branding-not-found' });
+        return res.ok({ branding: brand });
+      } catch(e:any) {
+        const { status, body } = mapError(e);
+        return res.status(status).json(body);
+      }
+    }
+
+    /** 9.2 Save draft variables */
+  async draft(req: Request, res: Response) {
+      const branding = req.params['branding'];
+      try {
+    const actor = (req as any).user || (req as any).session?.user; // policy should guarantee admin
+        const variables = req.body?.variables || {};
+        const updated = await BrandingService.saveDraft({ branding, variables, actor });
+        return res.ok({ branding: updated });
+      } catch(e:any) {
+        const { status, body } = mapError(e);
+        return res.status(status).json(body);
+      }
+    }
+
+    /** 9.3 Create preview token */
+  async preview(req: Request, res: Response) {
+      const branding = req.params['branding'];
+      const portal = req.params['portal'];
+      try {
+    const actor = (req as any).user || (req as any).session?.user;
```

Comment:
Why do these need to be cast to `any`? Does the request not have `user` or `session` properties?